### DISCUSSION TITLE
How to compute minStaveWidth for elements that i want to draw in stave?

### DISCUSSION THREAD

**Discussion by FRutkowski** [0xfe/vexflow#1464] [Q&A] [answered]
Im creating app using VexFlow for composing music. The problem i have is that when i add new note to my stave(to array where i keep the notes) i want to compute minimal stave width - because if i have too many notes in stave(bar) then they draw beyond the endBarLine. Computed minStaveWidth i will use later in  ```formatter.joinVoices([voices]).format([voices], /*place for minStaveWidth/*)```. Do you have an idea what is the easiest way to get this? Thank you for any help:)

**Comment by FRutkowski** [comment, marked as answer]
Okay if some one need to know how to do this, the way that worked for me was a create new instance of Formatter(), and use function ```preCalculateMinTotalWidth(voices)``` which returned ```minTotalWidth``` without using format function, if you want to use for every stave same instance of ```Formatter``` you'll probably need to change ```hasMinTotalWidth``` to ```false``` but i didnt test this.

**Comment by FRutkowski** [comment]
![image](https://user-images.githubusercontent.com/65611113/197543016-b31d2653-e175-4f34-8ea5-85b2708be1ad.png)
im not sure if there shouldnt be ```formatter.preCalculateMinTotalWidth()``` instead of ```Formatter.getMinTotalWidth()``` because second option is only a getter where a first option if you dont have minTotalWidth will calculate this for you

**Comment by AaronDavidNewman** [comment]
Hi, you are correct there is no way to estimate width within vexflow other than preCalculateMinTotalWidth.  hasMinTotalWidth is reset when you call joinVoices.  There doesn't appear to be another way to reset hasMinTotalWidth.  Kind of a hack, but it looks like you can always call joinVoices with an empty array to reset it.
